lib/src/io/node.dart

```diff
@@ -80,6 +100,19 @@ _readFile(String path, [String encoding]) {
   }
 }
 
+String readStdin() async {
+  var completer = new Completer<Null>();
+  var contents = new StringBuffer();
+  _stdin.on('data', allowInterop((buf) {
+    contents.write(UTF8.decode(buf));
+  }));
+  _stdin.on('end', allowInterop(() {
+    completer.complete();
+  }));
+  await completer.future;
```

Comment:
Rather than doing this dance, just make it a `Completer<String>` and return `completer.future`.